cmd/snap-recovery/partition/sfdisk.go

```diff
@@ -216,10 +217,10 @@ func buildPartitionList(ptable *sfdiskPartitionTable, pv *gadget.LaidOutVolume)
 			s.Size/sectorSize, partitionType(ptable.Label, p.Type), s.Name)
 
 		// Are roles unique so we can use it to map nodes? Should we use labels instead?
-		deviceMap[s.Role] = node
+		toBeCreated[node] = p
```

Comment:
Wish we had more reliable name to identify device nodes rather than making up the name ourselves. In the context of the followup PR, perhaps we should create, list the partitions and match then with structures, and then operate on the device nodes.